---
title: "First Project"
author: "Dima"
date: "May 17, 2019"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Introduction

This exercise is going to be a practice of R programming language and R-markdown tool. 

For this first practical, my data set is going to be taken from **GSE122020** study. Whole information about it can be accessed via this link: [link](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122020)

#Summary

Some brief information about the data and the study:

**Title:**  *Gene expression data from remote ischemic postconditioning in rat model of ischemia reperfusion injury*

**Organism:** *Rattus norvegicus*

**Experiment type:** *Expression profiling by array (microarray)*

####Summary of the project: 
Restoration of blood flow is the definitive therapy to salvage myocardium following ischemic injury. However, sudden restoration of blood flow to the ischemic myocardium causes ischemia reperfusion injury (IRI).
Here, the cardioprotective effect of remote ischemic postconditioning (RPostC) was investigated, based on our in vitro rat model of myocardial IRI. Three groups, including Sham, IRI, and IRI+ RPostC, were utilized for the analysis of Affymetrix Rat Gene 2.0 ST chip.

####Project's overall design 
Total RNAs were extracted and purified from the heart tissues of three groups using TRIzol reagent (Life Technologies) and RNeasy mini kit (Qiagen). Next, we synthesized the cDNA using an Ambion WT Expression Kit (Affymetrix), and obtained the cRNA by using a GeneChip WT Terminal Labeling and Controls Kit (Affymetrix, Inc.). Affymetrix Human Transcriptome Array 2.0 was then utilized to hybridize the fragmented cRNA product. After washing and staining in the GeneChip Fluidics Station 450, AffymetrixÂ® GeneChip Command Console (AGCC) and robust multichip analysis (RMA) algorithms were used to scan the arrays and analyze the data
 	
**Groups of observations:** *IRI. RpostC, control, each one of the groups has 3 samples inside of it, making a total of 9 samples*
 	
 	
####Contributors and citations:
*Wang Z, Wei M* 	

Wang Z, Wen J, Zhou C, Wang Z et al. Gene expression profiling analysis to investigate the role of remote ischemic postconditioning in ischemia-reperfusion injury in rats. BMC Genomics 2019 May 9;20(1):361. PMID: 31072368
 	
 	
 	
 	
#Data descriptiom

####Summary of the data 
```{r mat,echo=FALSE}
matt<-read.csv('gse122020_matrix_IDasInd.csv',row.names = 'X')
```
```{r mat,echo=FALSE}
summary(matt)
```
From this command we can see that there is not too much differences between the samples, so we can check it using boxplots
####Boxploting data to visualize it
```{r}
boxplot(matt,col=c(12,12,12,54,54,54,69,69,69))
```
And boxplot verify this information

Next steo is to visualize each sample with a histogram and check how the distribution of frequencies are
####Creating a histogram for each column to visualize each sample individualy
```{r echo=TRUE}

for (col in 1:ncol(matt)) {
    hist(matt[,col], xlab=)
}
```
We can see that first 6 samples are more distinct than the last 3, which are the control, lets see the means of the samples
####Getting the means of each column 
```{r echo=FALSE}
colMeans(matt)
```
We can see that, as expected, controls are moresimilar between each other, than other samples. We can now check if the samples fit the controls

```{r echo=FALSE}
chisq.test(c(matt$GSM3453058,matt$GSM3453064))
chisq.test(c(matt$GSM3453058,matt$GSM3453065))
chisq.test(c(matt$GSM3453058,matt$GSM3453066))

```




