---
title: "Final Microarray data analysis done by Dmytro Pravdyvets"
output: html_document
---
# Project summary
This is the final project of the second part of the course of Omics carried out by Alex Sanchez Pla, where the main goal is maily how to analyse Microarray data, learn how to obtain it and leaning how to use R and Rmd. The project consists of a report of an analysis of Microaray data of choice, which in my case is GSE114556. 

## Summary of the data 	
The obligate intracellular bacterium Chlamydia trachomatis replicates in a cytosolic vacuole in human epithelial cells. Infection of human cells with C. trachomatis causes substantial changes to many host cell signalling pathways but the molecular basis of such influence is not well understood. Studies of gene transcription of the infected cell have shown altered transcription of many host cell genes, indicating a transcriptional response of the host cell to the infection. We here describe that infection of human cells with C. trachomatis as well as infection of murine cells with C. muridarum profoundly inhibits protein synthesis of the infected host cell. This inhibition was accompanied by changes to the ribosomal profile of the infected cell indicative of a block of translation initiation, most likely as part of a stress response. The chlamydial protease CPAF also reduced protein synthesis in uninfected cells although CPAF-deficient C. trachomatis showed no defect in this respect. Analysis of polysomal mRNA as a proxy of actively transcribed mRNA identified a number of biological processes differentially affected by chlamydial infection. Mapping of differentially regulated genes onto a protein interaction network identified nodes of up- and down-regulated networks during chlamydial infection. Proteomic analysis of protein synthesis further suggested translational regulation of host cell functions by chlamydial infection. These results demonstrate reprogramming of the host cell during chlamydial infection through the alteration of protein synthesis.
 	
## Overall design	
HeLa cells were infected with C. trachomatis L2 for 33h or left untreated and polysomal as well as total RNA was isolated.

## Citations
Ohmer M, Tzivelekidis T, Niedenf√ºhr N, Volceanov-Hahn L et al. Infection of HeLa cells with Chlamydia trachomatis inhibits protein synthesis and causes multiple changes to host cell pathways. Cell Microbiol 2019 Apr;21(4):e12993. PMID: 30551267


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
Loading all the needed packages before running the analysis
```{r, include=FALSE}
if (!require(BiocManager)) install.packages("BiocManager")

installifnot <- function (pkg){
  if (!require(pkg, character.only=T)){
    BiocManager::install(pkg)
}else{
  require(pkg, character.only=T)
  }
}

installifnot("pd.mogene.1.0.st.v1")
installifnot("mogene10sttranscriptcluster.db")
installifnot("oligo")
installifnot("limma")
installifnot("Biobase")
installifnot("arrayQualityMetrics")
installifnot("genefilter")
installifnot("multtest")
installifnot("annotate")
installifnot("xtable")
installifnot("gplots")
installifnot("scatterplot3d")
```
#### targets summary
We can see that the whole grou of observation can be divided into 12 samples, that can be grouped into 4 different subgroups, the division into subgroups will be later used as factors for regression model
```{r}
targets <- read.csv('./data/targets.csv', header = T,sep = ';')
targets
```
```{r,include=FALSE}
workingDir <-getwd()
dataDir <- file.path(workingDir, "data")
resultsDir <- file.path(workingDir, "results")
setwd(resultsDir)
```
Loadign all the CEL files of the study that will be used for the analysis
```{r, include=F}
CELfiles <- list.celfiles(file.path(dataDir))
rawData <- read.celfiles(file.path(dataDir,CELfiles))

```
Defining 2 varaibles of the targets file that will be used for plots later on
```{r}
sampleNames <- as.character(targets$ShortNAme)
sampleColor <- as.character(targets$Colors)
```
### Quality control of arrays with raw data
We begin to analyse data with a boxplot 
```{r}
boxplot(rawData, which="all",las=2, main="Intensity distribution of RAW data", 
        cex.axis=0.6, col=sampleColor, names=sampleNames)
```
We can see that there not seem to be any artifacts with the data, but the distribution of the values seems to be somewhat different between the groups.

Next step would be to cluster the data
```{r}
clust.euclid.average <- hclust(dist(t(exprs(rawData))),method="average")
plot(clust.euclid.average, labels=sampleNames, main="Hierarchical clustering of RawData", 
     cex=0.7,  hang=-1)
```
It looks like there are some clear clusters of data if we compare the clusters to the boxplots, similar boxplot end up in the same cluster ex(69,72,73).

Now we do the principar component analysis
```{r}
plotPCA <- function ( X, labels=NULL, colors=NULL, dataDesc="", scale=FALSE, formapunts=NULL, myCex=0.8,...)
{
  pcX<-prcomp(t(X), scale=scale) # o prcomp(t(X))
  loads<- round(pcX$sdev^2/sum(pcX$sdev^2)*100,1)
  xlab<-c(paste("PC1",loads[1],"%"))
  ylab<-c(paste("PC2",loads[2],"%"))
  if (is.null(colors)) colors=1
  plot(pcX$x[,1:2],xlab=xlab,ylab=ylab, col=colors, pch=formapunts, 
       xlim=c(min(pcX$x[,1])-100000, max(pcX$x[,1])+100000),ylim=c(min(pcX$x[,2])-100000, max(pcX$x[,2])+100000))
  text(pcX$x[,1],pcX$x[,2], labels, pos=3, cex=myCex)
  title(paste("Plot of first 2 PCs for expressions in", dataDesc, sep=" "), cex=0.8)
}

plotPCA(exprs(rawData), labels=sampleNames, dataDesc="raw data", colors=sampleColor,
        formapunts=c(rep(16,4),rep(17,4)), myCex=0.6)

```
By doing this we try to reduce the size of our datasaet without loosing too much information (the principal of PCA itself basically), we can see which PC (1 or 2)
has more effect on a specific sample. 

###Data normalization and analysis
Now we normalize our data and check if the results are now different or same and if the normalization is okay without artifacts. 

```{r, include=F}
eset<-rma(rawData)

write.exprs(eset, file.path(resultsDir, "NormData.txt"))

```
Once data is normalized, we repeat the analysis done before and compare the results
```{r}
#BOXPLOT
boxplot(eset, las=2, main="Intensity distribution of Normalized data", cex.axis=0.6, 
        col=sampleColor, names=sampleNames)

```
Now we can see that the data has barely any difference, being someone with little experience in this field, I do not have a concrete answer, but after looking up things on internet and just intuevly, the fact that the data is so stabilized now is strange.

Next step is to cluster the observations
```{r}
clust.euclid.average <- hclust(dist(t(exprs(eset))),method="average")
plot(clust.euclid.average, labels=sampleNames, main="Hierarchical clustering of NormData", 
     cex=0.7,  hang=-1)
```
We can compare the obtained tree with the non-normalized data, and we can see few changes in the nodes, for example node 74 and 80 are not together now, 75 and 76 are not longer in the same cluster neither, but besides that other nodes are stil the same

Last step is to again analyze principal components
```{r}
plotPCA <- function ( X, labels=NULL, colors=NULL, dataDesc="", scale=FALSE, formapunts=NULL, myCex=0.8,...)
{
  pcX<-prcomp(t(X), scale=scale) # o prcomp(t(X))
  loads<- round(pcX$sdev^2/sum(pcX$sdev^2)*100,1)
  xlab<-c(paste("PC1",loads[1],"%"))
  ylab<-c(paste("PC2",loads[2],"%"))
  if (is.null(colors)) colors=1
  plot(pcX$x[,1:2],xlab=xlab,ylab=ylab, col=colors, pch=formapunts, 
       xlim=c(min(pcX$x[,1])-10, max(pcX$x[,1])+10),ylim=c(min(pcX$x[,2])-10, max(pcX$x[,2])+10))
  text(pcX$x[,1],pcX$x[,2], labels, pos=3, cex=myCex)
  title(paste("Plot of first 2 PCs for expressions in", dataDesc, sep=" "), cex=0.8)
}

plotPCA(exprs(eset), labels=sampleNames, dataDesc="NormData", colors=sampleColor,
        formapunts=c(rep(16,4),rep(17,4)), myCex=0.6)

```
Here is where we can see more difference, some previous points differ a lot and because of normalization, the scale of the graph is smaller too, so the previous one was better for this

###Array quality matrices
Now we are going to filter our data and check it's quality
```{r}
arrayQualityMetrics(rawData,  reporttitle="QualityControl", force=TRUE)
annotation(eset) <- "org.Mm.eg.db"
eset_filtered <- nsFilter(eset, var.func=IQR,
         var.cutoff=0.75, var.filter=TRUE,
         filterByQuantile=TRUE)
```
After the filtering,the amount of genes that are out is:
```{r}
print(eset_filtered$filter.log$numLowVar)
```
And the rest of the genes with the characteristic of the dataset can be obtained from here:
```{r}
print(eset_filtered$eset)
```

####Differential expressed genes, regression and plots
Now the analysis itself is going to be done, using differentially expressed genes, regression models and plots
```{r}
treat <- targets$grupos
lev <- factor(treat, levels = unique(treat))
design <-model.matrix(~0+lev)
colnames(design)<-c("it","ct","ip","cp")
rownames(design) <- sampleNames
print(design)
```
Here you can see the design model of the data, which has 4 different Alphas where each alpha has 3 samples

Time to do comparisons!
```{r}
cont.matrix1 <- makeContrasts (
  itvsct = it-ct,
  itvsip = it-ip,
  itvscp = it-cp,
  ctvsip = ct-ip,
  ctvscp = ct-cp,
  ipvscp = ip-cp,
  levels=design)
comparison1 <- "Effect of Induction"
cont.matrix1
```

Now we have both matrices, next step is to fit the data to a model

```{r}
fit1 <- lmFit(eset_filtered$eset, design)
fit.main1 <- contrasts.fit(fit1, cont.matrix1)
fit.main1
```

Another good idea is to account for bayeses in the model, for this we can run this code:
```{r}
fit.main1 <- eBayes(fit.main1)
fit.main1
```

####Toptables of the genes
We already have the matrices, model and data, so we can get the toptables for each comparison, join the into 1 and do a volcano plot of this abomination
```{r}
topTab_itvsct <- topTable (fit.main1, number=nrow(fit.main1), coef="itvsct", adjust="fdr"); 
head(topTab_itvsct)
topTab_itvsip <- topTable (fit.main1, number=nrow(fit.main1), coef="itvsip", adjust="fdr"); head(topTab_itvsip)
topTab_itvscp <- topTable (fit.main1, number=nrow(fit.main1), coef="itvscp", adjust="fdr"); head(topTab_itvsct)
topTab_ctvsip <- topTable (fit.main1, number=nrow(fit.main1), coef="ctvsip", adjust="fdr"); head(topTab_ctvsip)
topTab_ctvscp <- topTable (fit.main1, number=nrow(fit.main1), coef="ctvscp", adjust="fdr"); head(topTab_ctvscp)
topTab_ipvscp <- topTable (fit.main1, number=nrow(fit.main1), coef="ipvscp", adjust="fdr"); head(topTab_ipvscp )

merge(topTab_itvscp[1:1,],topTab_itvsct[1:1,])

k<-list(topTab_itvscp[1:round(NROW(topTab_itvscp)*0.05),],topTab_itvsct[1:round(NROW(topTab_itvsct)*0.05),],topTab_itvsip[1:round(NROW(topTab_itvsip)*0.05),],topTab_ctvscp[1:round(NROW(topTab_ctvscp)*0.05),],topTab_ctvsip[1:round(NROW(topTab_ctvsip)*0.05),],topTab_ipvscp[1:round(NROW(topTab_ipvscp)*0.05),])

#topTab<-Reduce(merge,list(topTab_itvscp[1:round(NROW(topTab_itvscp)*0.05),],topTab_itvsct[1:round(NROW(topTab_itvsct)*0.05),],topTab_itvsip[1:round(NROW(topTab_itvsip)*0.05),],topTab_ctvscp[1:round(NROW(topTab_ctvscp)*0.05),],topTab_ctvsip[1:round(NROW(topTab_ctvsip)*0.05),],topTab_ipvscp[1:round(NROW(topTab_ipvscp)*0.05),]))
topTab<-do.call('rbind',list(topTab_itvscp[1:round(NROW(topTab_itvscp)*0.05),],topTab_itvsct[1:round(NROW(topTab_itvsct)*0.05),],topTab_itvsip[1:round(NROW(topTab_itvsip)*0.05),],topTab_ctvscp[1:round(NROW(topTab_ctvscp)*0.05),],topTab_ctvsip[1:round(NROW(topTab_ctvsip)*0.05),],topTab_ipvscp[1:round(NROW(topTab_ipvscp)*0.05),]))
topTab
```

```{r}
volcanoplot(fit.main1, highlight=10, names=fit.main1$ID, 
            main = paste("Differentially expressed genes", colnames(cont.matrix1), sep="\n"))
abline(v = c(-3, 3))


pdf(file.path(resultsDir,"Volcanos.pdf"))
volcanoplot(fit.main1, highlight = 10, names = fit.main1$ID, 
            main = paste("Differentially expressed genes", colnames(cont.matrix1), sep = "\n"))
abline(v = c(-3, 3))
dev.off()

```
#---------------------------------------------------------------------------------------------
###HEATMAP PLOTS
#---------------------------------------------------------------------------------------------

#PREPARE THE DATA
```{r}

my_frame <- data.frame(exprs(eset))
head(my_frame)
HMdata <- merge(my_frame, topTab, by.x = 0, by.y = 0)
rownames(HMdata) <- HMdata$Row.names
HMdata <- HMdata[, -c(1,10:15)]
head(HMdata)
HMdata2 <- data.matrix(HMdata, rownames.force=TRUE)
head(HMdata2)
write.csv2(HMdata2, file = file.path(resultsDir,"Data2HM.csv"))

#HEATMAP PLOT
my_palette <- colorRampPalette(c("blue", "red"))(n = 299)
ncol(my_frame)
heatmap.2(HMdata2,
          Rowv=TRUE,
          Colv=TRUE,
          main="HeatMap it.vs.ct FC>=3",
          scale="row",
          col=my_palette,
          sepcolor="white",
          sepwidth=c(0.05,0.05),
          cexRow=0.5,
          cexCol=0.9,
          key=TRUE,
          keysize=1.5,
          density.info="histogram",
          ColSideColors=sampleColor,
          tracecol=NULL,
          srtCol=30)

#EXPORT TO PDF FILE
pdf(file.path(resultsDir,"HeatMap InducedvsWT.pdf"))
heatmap.2(HMdata2,
          Rowv=TRUE,
          Colv=TRUE,
          main="HeatMap Induced.vs.WT FC>=3",
          scale="row",
          col=my_palette,
          sepcolor="white",
          sepwidth=c(0.05,0.05),
          cexRow=0.5,
          cexCol=0.9,
          key=TRUE,
          keysize=1.5,
          density.info="histogram",
          ColSideColors=c(rep("red",4),rep("blue",4)),
          tracecol=NULL,
          srtCol=30)
dev.off()
```
#---------------------------------------------------------------------------------------------
###DATA ANNOTATION
#---------------------------------------------------------------------------------------------


#BiocManager::install("pd.hugene.2.0.st")
#BiocManager::install("hugene20sttranscriptcluster.db")

```{r}
require(pd.hugene.2.0.st)
require(hugene20sttranscriptcluster.db)

columns(hugene20sttranscriptcluster.db)
probes_itvsct<-rownames(head(topTab_itvsct))
antot_itvsct<-select(hugene20sttranscriptcluster.db,probes_itvsct,
                     columns = c("ENTREZID","SYMBOL","GENENAME"))
anot_itvsct
```
#---------------------------------------------------------------------------------------------
#END OF SCRIPT
#---------------------------------------------------------------------------------------------
